costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -378,6 +383,18 @@ void Costmap2DROS::mapUpdateLoop(double frequency)
   if (frequency == 0.0)
     return;
 
+  if (map_update_thread_affinity_ >= 0)
+  {
+    if (setThreadAffinity(map_update_thread_affinity_))
```

Comment:
We may want to think about making a wrapper class in the framework library.  Just so we can use it in other places.  Could wait till we use it in one other place.